api/proto/objects/revset.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+package jj_api.objects;
+
+message Revset {
+  oneof kind {
+    // Should ideally only be used when the user types in their own revset.
+    // Otherwise, construct it using this proto.
+    string revset = 1;
```

Comment:
Does this resolve revset aliases? If it does, it implies the server will have to know how to read the user's config.

Reply:
I think it should. If an extension ever wants to take input from a user, I think that the user would likely have the assumption that the revset will work correctly with their aliases.

Reply:
I think there's a reasonable middle ground where we could take in a boolean `use_user_config`, and default it to False. I think that there should be a way for the APIs to use user config, but I think your point is quite reasonable, and so maybe we should just not do so by default.

Reply:
I've added `GlobalOptions` (similar to jj-cli's global options), which now has the field `use_user_config` and `extra_configs`.